File: Caesar Cipher/CaesarCipher.java
// https://www.youtube.com/watch?v=JtbKh_12ctg
// https://www.youtube.com/watch?v=na5rapg1XsI

public class CaesarCipher
{
    
    private static String encrypt(String plainText, int shiftValue){
        
        String cipherText = "";
        
        for(int index = 0; index < plainText.length(); index++){
            
            char currChar = plainText.charAt(index);
            if(currChar != ' '){
                
                char firstAlpha;
                if(Character.isUpperCase(currChar))
                    firstAlpha = 'A';
                else 
                    firstAlpha = 'a';
                
                int originalAlphaPosition = currChar - firstAlpha;
                int newAlphaPosition = (originalAlphaPosition + shiftValue) % 26;
                
                char encryptedAlpha = (char)(firstAlpha + newAlphaPosition);
                
                cipherText += encryptedAlpha;
                
            }
            else 
                cipherText += ' ';
        }
        
        return cipherText;
        
    }
    
    private static String decrypt(String cipherText, int shiftValue){
        return encrypt(cipherText, shiftValue);
    }
    
    public static void main(String[] args) {

	String plainText = "aBcDeFgHiJKLMNOPQRSTUVW xyz";

	String encrypted = encrypt(plainText, 3);
	String decrypted = decrypt(encrypted, 26 - 3);

	System.out.println("Plain Text: " + plainText);
	System.out.println("Encrypted Text: " + encrypted);
	System.out.println("Decrypted Text: " + decrypted);

    }
}

File: Caesar Cipher/CaesarCipher.py
def encrypt(pt,k):
    pt=pt.split()
    ct=""
    for i in pt:
        for j in i:
            alph=ord(j)+(k%26)
            if j.islower():
                if alph>ord('z'):
                    alph-=26
            else:
                if alph>ord('Z'):
                    alph-=26
            ct+=chr(alph) 
        ct+=" "
    return ct
    
def decrypt(ct,k):
    ct=ct.split()
    pt=""
    for i in ct:
        for j in i:
            alph=ord(j)-(k%26)
            if j.islower():
                if alph<ord('a'):
                    alph+=26
            else:
                if alph<ord('A'):
                    alph+=26
            pt+=chr(alph) 
        pt+=" "
    return pt    

print("Enter Plaintext: ",end="")
plaintext=input().strip()
print("Enter key: ",end="")
k=int(input())
ciphertext=encrypt(plaintext,k)
print("Ciphertext:",ciphertext)
decryptedtext=decrypt(ciphertext,k)
print("Plaintext:",decryptedtext)

